### PR TITLE
Remove private system properties in jdk17+

### PR DESCRIPTION
**Pull request by pshipton**
Fixes https://github.com/eclipse-openj9/openj9/issues/17248

### CODE REVIEW

File: jcl/src/java.base/share/classes/java/lang/System.java [hunk 1]
```diff
@@ -805,6 +805,7 @@ private static void ensureProperties(boolean isInitialization) {
 	jdk.internal.util.StaticProperty.jdkSerialFilter();
 }
 
+/*[IF JAVA_SPEC_VERSION >= 12]*/
```
Comment (keithc-ca) [inline jcl/src/java.base/share/classes/java/lang/System.java:808]: I'd prefer this say `JAVA_SPEC_VERSION > 11` (or if you like, `JAVA_SPEC_VERSION >= 17`) so it makes reference to a LTS version.

File: jcl/src/java.base/share/classes/java/lang/System.java [hunk 2]
```diff
@@ -813,9 +814,20 @@ private static void ensureProperties(boolean isInitialization) {
...
 	for (Map.Entry<String, String> property : mapProperties.entrySet()) {
-		systemProperties.put(property.getKey(), property.getValue());
+		String key = property.getKey();
+		/* Remove OpenJDK private properties that should not be System properties. */
+		switch (key) {
+			case "sun.nio.MaxDirectMemorySize": //$NON-NLS-1$
+			case "sun.nio.PageAlignDirectMemory": //$NON-NLS-1$
+			case "java.lang.Integer.IntegerCache.high": //$NON-NLS-1$
+			case "sun.java.launcher.diag": //$NON-NLS-1$
+			case "jdk.boot.class.path.append": //$NON-NLS-1$
```
Comment (keithc-ca) [inline jcl/src/java.base/share/classes/java/lang/System.java:824]: Please list the cases alphabetically and add an explicit `default` case.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: jcl/src/java.base/share/classes/java/lang/System.java (hunk 2 above)
Comment (keithc-ca) [reply jcl/src/java.base/share/classes/java/lang/System.java:824]: Also, by convention, each `case` should be indented the same as `switch`.

Comment (pshipton) [reply jcl/src/java.base/share/classes/java/lang/System.java]: Addressed all the review comments.

**Review by keithc-ca — APPROVED** [review]

**Comment by keithc-ca** [comment]
Jenkins test sanity alinux64 jdk11,jdk17

**Comment by keithc-ca** [comment]
Some of the testing on aarch64_linux failed:
```
16:09:34  git fetch -q --tags https://github.com/eclipse-openj9/openj9.git +refs/pull/*:refs/remotes/origin/pr/*
16:09:43  error: RPC failed; curl 16 Error in the HTTP2 framing layer
16:09:43  fatal: the remote end hung up unexpectedly
```
Testing restarted.

**Comment by keithc-ca** [comment]
The test failure appears to be a known problem (https://github.com/eclipse-openj9/openj9/issues/13577).
```
Testing: -Xlockword minimizeFootprint mode
Test start time: 2023/04/24 22:34:05 Coordinated Universal Time
Running command: "/home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_aarch64_linux_Personal_testList_1/openjdkbinary/j2sdk-image/bin/java"   -Xdump -cp "/home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_aarch64_linux_Personal_testList_1/aqa-tests/TKG/../../jvmtest/functional/cmdLineTests/utils/utils.jar" -Xlockword:mode=minimizeFootprint VMBench.FibBench
Time spent starting: 4 milliseconds
Time spent executing: 3456 milliseconds
Test result: FAILED
Output from test:
 [ERR] Unhandled exception
 [ERR] Type=Segmentation error vmState=0x00000000
 [ERR] J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000001
 [ERR] Handler1=0000FFFFAC3F2514 Handler2=0000FFFFAC357AC0 InaccessibleAddress=0000000000000024
 [ERR] R0=00000000FFE77E44 R1=0000000000000004 R2=0000000000000005 R3=0000000000000003
 [ERR] R4=0000000000000000 R5=0000FFFFA80B0388 R6=0000000000000001 R7=000000000002B200
 [ERR] R8=0000FFFF51459707 R9=FFFFFFFFFFFE0A62 R10=000000000000001E R11=0000000000000008
 [ERR] R12=0000000000000018 R13=0000000000000003 R14=0000FFFFAC97D490 R15=0000000000000004
 [ERR] R16=0000000000000001 R17=0000FFFFACA51680 R18=0000000000000000 R19=0000000000011200
 [ERR] R20=00000000000F6B90 R21=0000000000000000 R22=00000000000F6B68 R23=00000000FFE77E40
 [ERR] R24=FFFFFFFFFFFFFFFF R25=0000000000000000 R26=0000FFFFA8024F50 R27=0000FFFFA8024F50
 [ERR] R28=0000FFFFAC4454F0 R29=0000FFFFAC97D2C0 R30=0000FFFFAC446A78 R31=0000FFFFAC97D2B0
 [ERR] PC=0000FFFFAC456E44 SP=0000FFFFAC97D2B0 PSTATE=0000000040001000
 [ERR] V0 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V1 ffffffffffffffff (f: 4294967296.000000, d: -nan)
 [ERR] V2 1fb6b9601b3111f9 (f: 456200704.000000, d: 6.620479e-156)
 [ERR] V3 0000ffffa814b9b0 (f: 2819930624.000000, d: 1.390664e-309)
 [ERR] V4 bfd00ea348b88334 (f: 1220051712.000000, d: -2.508934e-01)
 [ERR] V5 3fd5575b0be00b6a (f: 199232368.000000, d: 3.334568e-01)
 [ERR] V6 3fe62e42fefa39ef (f: 4277811712.000000, d: 6.931472e-01)
 [ERR] V7 4036000000000000 (f: 0.000000, d: 2.200000e+01)
 [ERR] V8 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V9 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V10 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V11 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V12 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V13 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V14 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V15 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V16 c00960cbbf788d5c (f: 3212348672.000000, d: -3.172264e+00)
 [ERR] V17 0000000000000800 (f: 2048.000000, d: 1.011846e-320)
 [ERR] V18 4000000000000000 (f: 0.000000, d: 2.000000e+00)
 [ERR] V19 3f9eb851eb851eb8 (f: 3951369984.000000, d: 3.000000e-02)
 [ERR] V20 3fb1eb851eb851ec (f: 515396064.000000, d: 7.000000e-02)
 [ERR] V21 0000000000000008 (f: 8.000000, d: 3.952525e-323)
 [ERR] V22 3f0000003f800000 (f: 1065353216.000000, d: 3.051759e-05)
 [ERR] V23 3fc999999999999a (f: 2576980480.000000, d: 2.000000e-01)
 [ERR] V24 3fd6666666666666 (f: 1717986944.000000, d: 3.500000e-01)
 [ERR] V25 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V26 3fb999999999999a (f: 2576980480.000000, d: 1.000000e-01)
 [ERR] V27 000000000000000a (f: 10.000000, d: 4.940656e-323)
 [ERR] V28 0000000000000800 (f: 2048.000000, d: 1.011846e-320)
 [ERR] V29 0000000000000300 (f: 768.000000, d: 3.794424e-321)
 [ERR] V30 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V31 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] Module=/home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_aarch64_linux_Personal_testList_1/openjdkbinary/j2sdk-image/lib/default/libj9vm29.so
 [ERR] Module_base_address=0000FFFFAC3C0000
 [ERR] Target=2_90_20230424_401 (Linux 5.15.0-69-generic)
 [ERR] CPU=aarch64 (8 logical CPUs) (0x1ef6b3000 RAM)
 [ERR] ----------- Stack Backtrace -----------
 [ERR] bytecodeLoopCompressed+0x11954 (0x0000FFFFAC456E44 [libj9vm29.so+0x96e44])
 [ERR] c_cInterpreter+0x54 (0x0000FFFFAC4CA7E4 [libj9vm29.so+0x10a7e4])
 [ERR] ---------------------------------------
```

**Pull request merged**